### PR TITLE
Use Webpack and React to render specific forms

### PR DESCRIPTION
**Pull request by JayDancer**
**GOAL**

- Display only the form fields relevant to chosen payment method in the checkout form.

**MEANS**

- Use Webpacker and React to display components from the PayTypeSelector .jsx files under javascript > PayTypeSelector.

### CODE REVIEW

File: app/javascript/PayTypeSelector/CheckPayType.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          <input type="text"
+                 name="order[account_number]"
+                 id="order_account_number" />
+        </div>
+      </div>
+    );
+  }
+}
+
+export default CheckPayType
```
Comment (schlaBAM) [inline app/javascript/PayTypeSelector/CheckPayType.jsx]: EOF

File: app/javascript/PayTypeSelector/CreditCardPayType.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                 id="order_expiration_date"
+                 size="9"
+                 placeholder="e.g/ 03/19" />
+        </div>
+      </div>
+    );
+  }
+}
+
+export default CreditCardPayType
```
Comment (schlaBAM) [inline app/javascript/PayTypeSelector/CreditCardPayType.jsx]: EOF

File: app/javascript/PayTypeSelector/NoPayType.jsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+class NoPayType extends React.Component {
+  render() {
+    return (<div></div>);
+  }
+}
+
+export default NoPayType
```
Comment (schlaBAM) [inline app/javascript/PayTypeSelector/NoPayType.jsx]: EOF

File: app/javascript/PayTypeSelector/PurchaseOrderPayType.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+          <input type="password"
+                 name="order[po_number]"
+                 id="order_po_number" />
+        </div>
+      </div>
+    );
+  }
+}
+
+export default PurchaseOrderPayType
```
Comment (schlaBAM) [inline app/javascript/PayTypeSelector/PurchaseOrderPayType.jsx]: EOF

File: app/javascript/PayTypeSelector/index.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        </div>
+        <PayTypeCustomComponent/>
+      </div>
+    );
+  }
+  onPayTypeSelected(event) {
+    this.setState({ selectedPayType: event.target.value });
+  }
+}
+export default PayTypeSelector
```
Comment (schlaBAM) [inline app/javascript/PayTypeSelector/index.jsx]: EOF

File: app/javascript/packs/pay_type.jsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+import PayTypeSelector from 'PayTypeSelector'
+
+// Ensures the code only runs after entire DOM loads.
+// Turbolinks makes sure that this is set back up every time
+// the page loads, rather than breaking subsequent loads.
+document.addEventListener('turbolinks:load', function() {
+  let element = document.getElementById("pay-type-component");
+  ReactDOM.render(<PayTypeSelector/>, element);
+});
```
Comment (schlaBAM) [inline app/javascript/packs/pay_type.jsx]: EOF

**Comment by JayDancer** [comment]
Good point, thank you! Added all the EOF lines :D I forgot to re-add Rubocop when we got these macbooks. Migrations are fine right now, have had to reset the db a few times and re-seed it, but doing alright; I'll keep that in mind!

**Pull request merged**